bin/all_sky_search/pycbc_bin_trigger_rates_dq

```diff
@@ -103,108 +109,103 @@ logging.info("Applying %s > %.3f cut", args.sngl_ranking,
-    del dq_data
-
-n_bins = args.n_time_bins
-percentiles = np.linspace(0, 100, n_bins+1)
-bin_times = np.zeros(n_bins)
-dq_percentiles = np.percentile(dq_logl, percentiles)[1:]
-
-# seconds bin tells what bin each second ends up
-seconds_bin = np.array([n_bins - np.sum(dq_percentiles >= dq_ll)
-                        for dq_ll in dq_logl]).astype(np.int64)
-del dq_percentiles
-
-# bin times tells how much time ends up in each bin
-bin_times = np.array([np.sum(seconds_bin == i)
-                      for i in range(n_bins)]).astype(np.float64)
-full_time = float(len(seconds_bin))
-times_nz = (bin_times > 0)
-del dq_logl
-
-# create a dict to look up dq percentile at any time
-dq_percentiles_time = dict(zip(dq_times, seconds_bin/n_bins))
-del dq_times
-
-with h5.File(args.bank_file, 'r') as bank:
-    if args.background_bins:
-        logging.info('Sorting bank into bins')
-        data = {
-            'mass1': bank['mass1'][:],
-            'mass2': bank['mass2'][:],
-            'spin1z': bank['spin1z'][:],
-            'spin2z': bank['spin2z'][:],
-            'f_lower': np.ones_like(bank['mass1'][:]) * args.f_lower
-            }
-        locs_dict = pycbc.events.background_bin_from_string(
-                                                   args.background_bins, data)
-        del data
-        locs_names = [b.split(':')[0] for b in args.background_bins]
-    else:
-        locs_dict = {'all_bin': np.arange(0, len(bank['mass1'][:]), 1)}
-        locs_names = ['all_bin']
-
-if args.prune_number > 0:
-    for bin_name in locs_names:
-        logging.info('Pruning bin %s', bin_name)
-        bin_locs = locs_dict[bin_name]
-        inbin = np.isin(tmplt_ids, bin_locs)
-        trig_times_bin = trig_times[inbin]
-        trig_stats_bin = stat[inbin]
-
-        for j in range(args.prune_number):
-            max_stat_arg = np.argmax(trig_stats_bin)
-            remove = abs(trig_times_bin[max_stat_arg] - trig_times) \
-                         < args.prune_window
-            logging.info("Prune %d: pruning %d triggers", j, sum(remove))
-            remove_inbin = abs(trig_times_bin[max_stat_arg]
-                               - trig_times_bin) < args.prune_window
-            stat[remove] = 0
-            trig_stats_bin[remove_inbin] = 0
-    keep = np.flatnonzero(stat)
-    logging.info("%d triggers removed through pruning", keep.size)
-    trig_times_int = trig_times_int[keep]
-    tmplt_ids = tmplt_ids[keep]
-    del stat
-    del keep
-
-del trig_times
-
+# Get the template bins
+bin_tids_dict = {}
+with h5.File(args.template_bins_file, 'r') as f:
+    ifo_grp = f[ifo]
+    for bin_name in ifo_grp.keys():
+        bin_tids_dict[bin_name] = ifo_grp[bin_name]['tids'][:]
+
+
+# get analysis segments
+analysis_segs = select_segments_by_definer(
+    args.analysis_segment_file,
+    segment_name=args.analysis_segment_name,
+    ifo=ifo)
+
+
+# utility function to compute segment list durations
+def segs_duration(segs):
```

Comment:
I don't think this is needed, segs.abs() should do the same calculation - see https://git.ligo.org/lscsoft/ligo-segments/-/blob/master/ligo/segments/__init__.py#L579